src/modules/endpoint/sagas/loadCategories.js

```diff
@@ -21,19 +22,21 @@ function * loadCategories (
   dataContainer: DataContainer,
   shouldUpdate: boolean
 ): Saga<FetchMapType> {
-  if (!dataContainer.categoriesAvailable() || shouldUpdate) {
+  const context = new DatabaseContext(city, language)
+
+  if (!dataContainer.categoriesAvailable(context) || shouldUpdate) {
```

Comment:
The problem now is -- since you completely changed the behavior of the DataContainer --, that `dataContainer.categoriesAvailable` (as well as all other `xyzAvailable` methods) will evaluate to false even if categories are available but not yet loaded from the database.
You might want to revert the DataContainer to that extent, that on every context switch all available data regarding that context is loaded.

Reply:
Also, why is this change in architecture necessary? Is there no other way to work with the current DataContainer?

Reply:
Should be fine now :)

Reply:
This is still a problem

Reply:
Should be solved finally :)

Reply:
Availability checks work now. I promise! :)